File: Layout/Layout_1st_order/Layout_1st_order_65nm/floorplan_script.m
%
fileID = fopen('floorplan.txt', 'w');

fprintf(fileID, '\n');
fprintf(fileID, 'selectObject Module I0 \n');
fprintf(fileID, 'createInstGroup PD10 -fence %.4f %.4f %.4f %.4f \n', loc_Rin_lef);
fprintf(fileID, 'setInstGroupPhyHier PD10 \n');
fprintf(fileID, 'addInstToInstGroup PD10 {I0 } \n');
fprintf(fileID, 'deselectModule I0 \n');
fprintf(fileID, '\n');

fprintf(fileID, 'selectObject Module I1 \n');
fprintf(fileID, 'createInstGroup PD9 -fence %.4f %.4f %.4f %.4f \n', loc_Rin_rig);
fprintf(fileID, 'setInstGroupPhyHier PD9 \n');
fprintf(fileID, 'addInstToInstGroup PD9 {I1 } \n');
fprintf(fileID, 'deselectModule I1 \n');
fprintf(fileID, '\n');

fprintf(fileID, 'selectObject Module I8 \n');
fprintf(fileID, 'createInstGroup PD8 -fence %.4f %.4f %.4f %.4f \n', loc_DAC_lef);
fprintf(fileID, 'setInstGroupPhyHier PD8 \n');
fprintf(fileID, 'addInstToInstGroup PD8 {I8 } \n');
fprintf(fileID, 'deselectModule I8 \n');
fprintf(fileID, '\n');

fprintf(fileID, 'selectObject Module I9 \n');
fprintf(fileID, 'createInstGroup PD7 -fence %.4f %.4f %.4f %.4f \n', loc_DAC_rig);
fprintf(fileID, 'setInstGroupPhyHier PD7 \n');
fprintf(fileID, 'addInstToInstGroup PD7 {I9 } \n');
fprintf(fileID, 'deselectModule I9 \n');
fprintf(fileID, '\n');

fprintf(fileID, 'selectObject Module I6 \n');
fprintf(fileID, 'createInstGroup PD6 -fence %.4f %.4f %.4f %.4f \n', loc_DIG);
fprintf(fileID, 'setInstGroupPhyHier PD6 \n');
fprintf(fileID, 'addInstToInstGroup PD6 {I6 } \n');
fprintf(fileID, 'deselectModule I6 \n');
fprintf(fileID, '\n');

fprintf(fileID, 'selectObject Module I7 \n');
fprintf(fileID, 'createInstGroup PD5 -fence %.4f %.4f %.4f %.4f \n', loc_DAC_BUF);
fprintf(fileID, 'setInstGroupPhyHier PD5 \n');
fprintf(fileID, 'addInstToInstGroup PD5 {I7 } \n');
fprintf(fileID, 'deselectModule I7 \n');
fprintf(fileID, '\n');

fprintf(fileID, 'selectObject Module I4 \n');
fprintf(fileID, 'createInstGroup PD4 -fence %.4f %.4f %.4f %.4f \n', loc_BUF_lef);
fprintf(fileID, 'setInstGroupPhyHier PD4 \n');
fprintf(fileID, 'addInstToInstGroup PD4 {I4 } \n');
fprintf(fileID, 'deselectModule I4 \n');
fprintf(fileID, '\n');

fprintf(fileID, 'selectObject Module I5 \n');
fprintf(fileID, 'createInstGroup PD3 -fence %.4f %.4f %.4f %.4f \n', loc_BUF_rig);
fprintf(fileID, 'setInstGroupPhyHier PD3 \n');
fprintf(fileID, 'addInstToInstGroup PD3 {I5 } \n');
fprintf(fileID, 'deselectModule I5 \n');
fprintf(fileID, '\n');

fprintf(fileID, 'selectObject Module I3 \n');
fprintf(fileID, 'createInstGroup PD1 -fence %.4f %.4f %.4f %.4f \n', loc_CCO_rig);
fprintf(fileID, 'setInstGroupPhyHier PD1 \n');
fprintf(fileID, 'addInstToInstGroup PD1 {I3 } \n');
fprintf(fileID, 'deselectModule I3 \n');
fprintf(fileID, '\n');

fprintf(fileID, 'selectObject Module I2 \n');
fprintf(fileID, 'createInstGroup PD2 -fence %.4f %.4f %.4f %.4f \n', loc_CCO_lef);
fprintf(fileID, 'setInstGroupPhyHier PD2 \n');
fprintf(fileID, 'addInstToInstGroup PD2 {I2 } \n');
fprintf(fileID, 'deselectModule I2 \n');
fprintf(fileID, '\n');

fclose(fileID);

File: Layout/Layout_1st_order_65nm_py/floorplan_script.py
with open('floorplan.txt', 'w') as fileID:
    fileID.write('\n')
    fileID.write('selectObject Module I0 \n')
    fileID.write('createInstGroup PD10 -fence {:.4f} {:.4f} {:.4f} {:.4f} \n'.format(*loc_Rin_lef))
    fileID.write('setInstGroupPhyHier PD10 \n')
    fileID.write('addInstToInstGroup PD10 {I0 } \n')
    fileID.write('deselectModule I0 \n')
    fileID.write('\n')

    fileID.write('selectObject Module I1 \n')
    fileID.write('createInstGroup PD9 -fence {:.4f} {:.4f} {:.4f} {:.4f} \n'.format(*loc_Rin_rig))
    fileID.write('setInstGroupPhyHier PD9 \n')
    fileID.write('addInstToInstGroup PD9 {I1 } \n')
    fileID.write('deselectModule I1 \n')
    fileID.write('\n')

    fileID.write('selectObject Module I8 \n')
    fileID.write('createInstGroup PD8 -fence {:.4f} {:.4f} {:.4f} {:.4f} \n'.format(*loc_DAC_lef))
    fileID.write('setInstGroupPhyHier PD8 \n')
    fileID.write('addInstToInstGroup PD8 {I8 } \n')
    fileID.write('deselectModule I8 \n')
    fileID.write('\n')

    fileID.write('selectObject Module I9 \n')
    fileID.write('createInstGroup PD7 -fence {:.4f} {:.4f} {:.4f} {:.4f} \n'.format(*loc_DAC_rig))
    fileID.write('setInstGroupPhyHier PD7 \n')
    fileID.write('addInstToInstGroup PD7 {I9 } \n')
    fileID.write('deselectModule I9 \n')
    fileID.write('\n')

    fileID.write('selectObject Module I6 \n')
    fileID.write('createInstGroup PD6 -fence {:.4f} {:.4f} {:.4f} {:.4f} \n'.format(*loc_DIG))
    fileID.write('setInstGroupPhyHier PD6 \n')
    fileID.write('addInstToInstGroup PD6 {I6 } \n')
    fileID.write('deselectModule I6 \n')
    fileID.write('\n')

    fileID.write('selectObject Module I7 \n')
    fileID.write('createInstGroup PD5 -fence {:.4f} {:.4f} {:.4f} {:.4f} \n'.format(*loc_DAC_BUF))
    fileID.write('setInstGroupPhyHier PD5 \n')
    fileID.write('addInstToInstGroup PD5 {I7 } \n')
    fileID.write('deselectModule I7 \n')
    fileID.write('\n')

    fileID.write('selectObject Module I4 \n')
    fileID.write('createInstGroup PD4 -fence {:.4f} {:.4f} {:.4f} {:.4f} \n'.format(*loc_BUF_lef))
    fileID.write('setInstGroupPhyHier PD4 \n')
    fileID.write('addInstToInstGroup PD4 {I4 } \n')
    fileID.write('deselectModule I4 \n')
    fileID.write('\n')

    fileID.write('selectObject Module I5 \n')
    fileID.write('createInstGroup PD3 -fence {:.4f} {:.4f} {:.4f} {:.4f} \n'.format(*loc_BUF_rig))
    fileID.write('setInstGroupPhyHier PD3 \n')
    fileID.write('addInstToInstGroup PD3 {I5 } \n')
    fileID.write('deselectModule I5 \n')
    fileID.write('\n')

    fileID.write('selectObject Module I3 \n')
    fileID.write('createInstGroup PD1 -fence {:.4f} {:.4f} {:.4f} {:.4f} \n'.format(*loc_CCO_rig))
    fileID.write('setInstGroupPhyHier PD1 \n')
    fileID.write('addInstToInstGroup PD1 {I3 } \n')
    fileID.write('deselectModule I3 \n')
    fileID.write('\n')

    fileID.write('selectObject Module I2 \n')
    fileID.write('createInstGroup PD2 -fence {:.4f} {:.4f} {:.4f} {:.4f} \n'.format(*loc_CCO_lef))
    fileID.write('setInstGroupPhyHier PD2 \n')
    fileID.write('addInstToInstGroup PD2 {I2 } \n')
    fileID.write('deselectModule I2 \n')
    fileID.write('\n')

